fortuna/src/Login.js

```diff
@@ -1,49 +1,108 @@
 //@flow strict
 import * as React from 'react';
+import { Link } from 'react-router-dom';
 import 'bootstrap/dist/css/bootstrap.min.css';
-import ReactDOM from 'react-dom';
-import MarketPlace from './MarketPlace.js';
+import LoginPopup from './LoginPopup.js';
 import './Login.css';
-  
+const PORT = 3000;
+
   //This is the Login Page Display
-  class LoginPageDisplay extends React.Component {
-    handleLoginClick = () => {
-        LoginFunction();
-      };
-    render(){
+  class Login extends React.Component<{||}> {
+    state = {
+      response: '',
+      userName: '',
+      password: '',
+      responseToPost: '',
+    };
```

Comment:
The <{||}> part of the `React.Component<{||}>` means that you have no props and no state. But you are using and relying on your state being not empty.

As a side note, it's probably a better idea to set the state in the constructor, as demoed on section 5.2 in the react docs here: https://reactjs.org/docs/state-and-lifecycle.html

The reason this is the case is because you are setting state like it is a class variable. This isn't a thing that should usually exist for components. The class variables could be updated, but the component won't re-render them.

Reply:
Ok I deleted the <{||}>. I don't really need these things to be re-rendered as I am never showing them to the user.

Reply:
You always want things to be rerendered; that's the main benefit of the React. Instead of deleteing the types entirely, you have to say what the types are. (namely, don't make your state and props empty, make them contain exactly the information you expect them to have)